File: money_exchange_rate/src/main/java/com/tools/money/network/CurrencyAPI.java
package com.tools.money.network;

import androidx.lifecycle.LiveData;

import com.github.leonardoxh.livedatacalladapter.Resource;

import java.util.Map;

import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.QueryMap;

public interface CurrencyAPI {

    @Headers({
            "x-rapidapi-host:currency-converter5.p.rapidapi.com",
            "x-rapidapi-key:xqHhD8Viw0mshdfXqdmyYwkpirccp1ZMhPejsnT7iBCSqthqK8",
            "format:json"})
    @GET("list")
    LiveData<Resource<CurrencyListResponse>> getCurrencyList();

    @Headers({
            "x-rapidapi-host:currency-converter5.p.rapidapi.com",
            "x-rapidapi-key:xqHhD8Viw0mshdfXqdmyYwkpirccp1ZMhPejsnT7iBCSqthqK8",
            "format:json"})
    @GET("convert")
    LiveData<Resource<ConvertRateResponse>> getConvertRate(@QueryMap Map<String, String> queryParams);

}

File: money_exchange_rate/src/main/java/com/tools/money/network/CurrencyAPI.kt
package com.tools.money.network

import com.tools.core.network.header.AuthenticatedHeaders
import retrofit2.http.GET
import retrofit2.http.HeaderMap
import retrofit2.http.Headers
import retrofit2.http.QueryMap

interface CurrencyAPI {
    @GET("/other/api/currencies/list")
    suspend fun getCurrencyList(@HeaderMap authenticatedHeaders: AuthenticatedHeaders): CurrencyListResponse

    @GET("/other/api/currencies/convert")
    suspend fun getConvertRate(@HeaderMap authenticatedHeaders: AuthenticatedHeaders, @QueryMap queryParams: Map<String, String>): ConvertRateResponse
}